### PR TITLE
Improve error message of vol_to_surf when input is wrong

### PR DESCRIPTION
**Pull request by himanshupathak21061998**
I closed previous pull request because I messed up with a lot of things to rebase master so I started from a fresh pull request @KamalakerDadi . Thanks for helping me.

### CODE REVIEW

File: nilearn/surface/surface.py [hunk 1]
```diff
@@ -619,27 +619,51 @@ def _gifti_img_to_mesh(gifti_img):
             coords = gifti_img.get_arrays_from_intent(
                 nibabel.nifti1.intent_codes['NIFTI_INTENT_POINTSET'])[0].data
         except IndexError:
-            raise ValueError('Gifti file needs to contain a data array '
-                             'with intent NIFTI_INTENT_POINTSET')
+            raise ValueError('The surf_mesh input is not recognized.'
+                             'Valid Freesurfer surface mesh inputs are ”pial”'
+                             'or “inflated” or “sphere” or “orig” or “white” '
+                             'which has intent code “NIFTI_INTENT_POINTSET”. '
```
Comment (bthirion) [inline nilearn/surface/surface.py]: ...which have intent code...

File: nilearn/surface/surface.py [hunk 2]
```diff
@@ -619,27 +619,51 @@ def _gifti_img_to_mesh(gifti_img):
...
         try:
             faces = gifti_img.get_arrays_from_intent(
                 nibabel.nifti1.intent_codes['NIFTI_INTENT_TRIANGLE'])[0].data
         except IndexError:
-            raise ValueError('Gifti file needs to contain a data array '
-                             'with intent NIFTI_INTENT_TRIANGLE')
+            raise ValueError('The surf_mesh input is not recognized.'
+                             'Valid Freesurfer surface mesh inputs are ”pial”'
+                             'or “inflated” or “sphere” or “orig” or “white” '
+                             'which has intent code “NIFTI_INTENT_POINTSET”. '
```
Comment (bthirion) [inline nilearn/surface/surface.py]: same here

File: nilearn/surface/surface.py [hunk 3]
```diff
@@ -619,27 +619,51 @@ def _gifti_img_to_mesh(gifti_img):
...
         try:
             coords = gifti_img.getArraysFromIntent(
                 nibabel.nifti1.intent_codes['NIFTI_INTENT_POINTSET'])[0].data
         except IndexError:
-            raise ValueError('Gifti file needs to contain a data array '
-                             'with intent NIFTI_INTENT_POINTSET')
+            raise ValueError('The surf_mesh input is not recognized.'
+                             'Valid Freesurfer surface mesh inputs are ”pial”'
+                             'or “inflated” or “sphere” or “orig” or “white” '
+                             'which has intent code “NIFTI_INTENT_POINTSET”. '
```
Comment (bthirion) [inline nilearn/surface/surface.py]: same here

File: nilearn/surface/surface.py [hunk 4]
```diff
@@ -619,27 +619,51 @@ def _gifti_img_to_mesh(gifti_img):
...
         try:
             faces = gifti_img.getArraysFromIntent(
                 nibabel.nifti1.intent_codes['NIFTI_INTENT_TRIANGLE'])[0].data
         except IndexError:
-            raise ValueError('Gifti file needs to contain a data array '
-                             'with intent NIFTI_INTENT_TRIANGLE')
+            raise ValueError('The surf_mesh input is not recognized.'
+                             'Valid Freesurfer surface mesh inputs are ”pial”'
+                             'or “inflated” or “sphere” or “orig” or “white” '
+                             'which has intent code “NIFTI_INTENT_POINTSET”. '
```
Comment (bthirion) [inline nilearn/surface/surface.py]: same here

File: nilearn/surface/surface.py [hunk 5]
```diff
@@ -619,27 +619,51 @@ def _gifti_img_to_mesh(gifti_img):
             coords = gifti_img.get_arrays_from_intent(
                 nibabel.nifti1.intent_codes['NIFTI_INTENT_POINTSET'])[0].data
         except IndexError:
-            raise ValueError('Gifti file needs to contain a data array '
-                             'with intent NIFTI_INTENT_POINTSET')
+            raise ValueError('The surf_mesh input is not recognized.'
```
Comment (KamalakerDadi) [inline nilearn/surface/surface.py]: As the same message is repeated multiple times we could write like this, **above** nibabel version check, something like this:

```python
error_message = ('The surf_mesh input is not recognized. Valid Freesurfer '
                 'surface mesh inputs are .pial, .inflated, .sphere, '
                 '.orig, .white. You provided input which has no '
                 '{0} or of empty value={1}')    

if LooseVersion(nibabel.__version__) >= LooseVersion('2.1.0'): 
```
 
Try to raise error message like this at relevant place pointing to NIFTI_INTENT_POINTSET:

```python
raise ValueError(error_message.format(
    'NIFTI_INTENT_POINTSET', gifti_img.get_arrays_from_intent(
    nibabel.nifti1.intent_codes['NIFTI_INTENT_POINTSET'])))
```


Like this pointing to NIFTI_INTENT_TRIANGLE
```python
raise ValueError(error_message.format(
    'NIFTI_INTENT_TRIANGLE', gifti_img.get_arrays_from_intent(
    nibabel.nifti1.intent_codes['NIFTI_INTENT_TRIANGLE'])))
```

**Comment by KamalakerDadi** [comment]
Attempts to close issue #1756

File: nilearn/surface/surface.py (hunk 5 above)
Comment (himanshupathak21061998) [reply nilearn/surface/surface.py]: Is it efficient  to assign a variable just to print an error message  but if you're saying it I will do it

Comment (KamalakerDadi) [reply nilearn/surface/surface.py]: Just to make it look clean. I don't better alternative. If you have you can propose.

Comment (kchawla-pi) [reply nilearn/surface/surface.py]: @himanshupathak21061998 If we don't assign a variable, then the information/value is assigned to an invisible, temporary variable in the background; the memory consumption is nearly the same, explicit variable or not.
Assigning it to a variable makes the code easier to parse for anyone who works with the code regularly and is considered a "Clean code" practice, industry-wide.

Comment (himanshupathak21061998) [reply nilearn/surface/surface.py]: OK I  will change it for a clean code thanks for suggestion .

**Comment by himanshupathak21061998** [comment]
Sorry for delay I was at my college that time . Thanks for giving suggestions

**Comment by KamalakerDadi** [comment]
I am only worried about the code cov report. It is showing 0.39% decrease without a major code refactoring.

Otherwise, LGTM. @GaelVaroquaux Does this error message sounds good ?

**Comment by GaelVaroquaux** [comment]
I think that the codecov thing is a fluke. This seems like it is good to merge, as far as I am concerned. Merging. Thank you!

**Pull request merged**